File: backend/app/schemas/company_connection.py
"""Pydantic schemas for company connections."""

from datetime import datetime

from pydantic import BaseModel, Field


class CompanyConnectionBase(BaseModel):
    """Base schema for company connection."""

    connection_type: str = Field(default="standard", max_length=50)
    can_message: bool = Field(default=True)
    can_view_profile: bool = Field(default=True)
    can_assign_tasks: bool = Field(default=False)


class UserToCompanyConnectionCreate(CompanyConnectionBase):
    """Schema for creating user-to-company connection."""

    target_company_id: int = Field(..., description="Target company ID")


class CompanyToCompanyConnectionCreate(CompanyConnectionBase):
    """Schema for creating company-to-company connection."""

    source_company_id: int = Field(..., description="Source company ID")
    target_company_id: int = Field(..., description="Target company ID")


class CompanyConnectionInfo(BaseModel):
    """Schema for company connection response."""

    id: int
    source_type: str
    source_user_id: int | None = None
    source_company_id: int | None = None
    target_company_id: int
    is_active: bool
    connection_type: str
    can_message: bool
    can_view_profile: bool
    can_assign_tasks: bool
    creation_type: str
    created_by: int | None = None
    created_at: datetime
    updated_at: datetime | None = None

    # Computed fields for display
    source_display_name: str | None = None
    target_company_name: str | None = None

    class Config:
        from_attributes = True


class CompanyConnectionUpdate(BaseModel):
    """Schema for updating company connection permissions."""

    connection_type: str | None = Field(None, max_length=50)
    can_message: bool | None = None
    can_view_profile: bool | None = None
    can_assign_tasks: bool | None = None

File: frontend/src/types/company-connection.ts
/**
 * Company connection types for company-based interactions
 */

export type CompanyConnectionSourceType = 'user' | 'company';

export interface CompanyConnection {
  id: number;
  source_type: CompanyConnectionSourceType;
  source_user_id: number | null;
  source_company_id: number | null;
  target_company_id: number;
  is_active: boolean;
  connection_type: string;
  can_message: boolean;
  can_view_profile: boolean;
  can_assign_tasks: boolean;
  creation_type: string;
  created_by: number | null;
  created_at: string;
  updated_at: string | null;
  source_display_name?: string;
  target_company_name?: string;
}

export interface UserToCompanyConnectionCreate {
  target_company_id: number;
  connection_type?: string;
  can_message?: boolean;
  can_view_profile?: boolean;
  can_assign_tasks?: boolean;
}

export interface CompanyToCompanyConnectionCreate {
  source_company_id: number;
  target_company_id: number;
  connection_type?: string;
  can_message?: boolean;
  can_view_profile?: boolean;
  can_assign_tasks?: boolean;
}

export interface CompanyConnectionUpdate {
  connection_type?: string;
  can_message?: boolean;
  can_view_profile?: boolean;
  can_assign_tasks?: boolean;
}

export interface CompanyConnectionFilters {
  source_type?: CompanyConnectionSourceType;
  is_active?: boolean;
  connection_type?: string;
}
